Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -512,6 +512,8 @@ void HTMLInputElement::updateType(const AtomString& typeAttributeValue)
         return;
     ASSERT(m_inputType->type() != newType->type());
 
+    Style::PseudoClassChangeInvalidation defaultInvalidation(*this, CSSSelector::PseudoClassType::Default, Style::PseudoClassChangeInvalidation::AnyValue);
```

Comment:
I think you can now do more fine grained invalidation than AnyValue with the new m_isDefaultChecked flag.

Reply:
:default can also match some other input types, and I didn't want to add extra logic for those. I guess I could've done the more fine grained invalidation by checking the specific types we're changing from / to but it didn't seem worth it.